.travis.yml

```diff
@@ -11,68 +11,72 @@ cache:
   directories:
     - $HOME/.composer/cache
 
-env:
-  global:
-    - EVENT_MANAGER_VERSION="^3.0"
-    - HYDRATOR_VERSION="^2.1"
-    - SERVICE_MANAGER_VERSION="^3.0.3"
-    - STDLIB_VERSION="^3.0"
-
 matrix:
   fast_finish: true
   include:
     - php: 5.5
       env:
-        - EXECUTE_CS_CHECK=true
+        - DEPS=lowest
+    - php: 5.5
+      env:
+        - DEPS=locked
     - php: 5.5
       env:
-        - EVENT_MANAGER_VERSION="^2.6.2"
-        - HYDRATOR_VERSION="^1.1"
-        - SERVICE_MANAGER_VERSION="^2.7.5"
-        - STDLIB_VERSION="^2.7.5"
+        - DEPS=latest
     - php: 5.6
       env:
-        - EXECUTE_TEST_COVERALLS=true
+        - DEPS=lowest
     - php: 5.6
       env:
-        - EVENT_MANAGER_VERSION="^2.6.2"
-        - HYDRATOR_VERSION="^1.1"
-        - SERVICE_MANAGER_VERSION="^2.7.5"
-        - STDLIB_VERSION="^2.7.5"
+        - DEPS=locked
+        - TEST_COVERAGE=true
+    - php: 5.6
+      env:
+        - DEPS=latest
+    - php: 7
+      env:
+        - DEPS=lowest
     - php: 7
+      env:
+        - DEPS=locked
+        - CHECK_CS=true
     - php: 7
       env:
-        - EVENT_MANAGER_VERSION="^2.6.2"
-        - HYDRATOR_VERSION="^1.1"
-        - SERVICE_MANAGER_VERSION="^2.7.5"
-        - STDLIB_VERSION="^2.7.5"
-    - php: hhvm 
-    - php: hhvm 
+        - DEPS=latest
+    - php: hhvm
       env:
-        - EVENT_MANAGER_VERSION="^2.6.2"
-        - HYDRATOR_VERSION="^1.1"
-        - SERVICE_MANAGER_VERSION="^2.7.5"
-        - STDLIB_VERSION="^2.7.5"
+        - DEPS=lowest
+    - php: hhvm
+      env:
+        - DEPS=locked
+    - php: hhvm
+      env:
+        - DEPS=latest
   allow_failures:
     - php: hhvm
 
+env:
+  global:
+    - COMPOSER_ARGS="--no-interaction --ignore-platform-reqs"
+
 notifications:
   irc: "irc.freenode.org#zftalk.dev"
   email: false
 
 before_install:
-  - if [[ $EXECUTE_TEST_COVERALLS != 'true' ]]; then phpenv config-rm xdebug.ini || return 0 ; fi
-  - composer self-update
-  - if [[ $EXECUTE_TEST_COVERALLS == 'true' ]]; then composer require --dev --no-update satooshi/php-coveralls ; fi
-  - composer require --no-update "zendframework/zend-eventmanager:$EVENT_MANAGER_VERSION" "zendframework/zend-servicemanager:$SERVICE_MANAGER_VERSION" "zendframework/zend-hydrator:$HYDRATOR_VERSION" "zendframework/zend-stdlib:$STDLIB_VERSION"
+  - travis_retry composer self-update
+  - if [[ $TRAVIS_PHP_VERSION != "hhvm" && $TEST_COVERAGE != 'true' ]]; then phpenv config-rm xdebug.ini ; fi
 
 install:
-  - travis_retry composer install --no-interaction --ignore-platform-reqs
+  - if [[ $DEPS == 'latest' ]]; then travis_retry composer update $COMPOSER_ARGS ; fi
+  - if [[ $DEPS == 'lowest' ]]; then travis_retry composer update --prefer-lowest --prefer-stable $COMPOSER_ARGS ; fi
+  - if [[ $TEST_COVERAGE == 'true' ]]; then travis_retry composer require --dev $COMPOSER_ARGS satooshi/php-coveralls ; fi
+  - travis_retry composer install $COMPOSER_ARGS
```

Comment:
That is intended as a "catch-all" scenario: the updates are only executed in special cases, and executing an install right after an update is basically a no-op. Since every line of the `install` block a different build environment, I preferred to not enclose this one in a conditional block to make it stand out as the lowest common denominator.